atef/widgets/happi.py

```diff
@@ -345,18 +358,25 @@ def _update_metadata(self):
         if self.client is None or self.item_name is None:
             return
 
-        try:
-            self.item = self.client[self.item_name]
-        except KeyError:
-            self.item = None
+        def get_metadata():
+            if self.client is None:
+                return {}
+
+            try:
+                self.item = self.client[self.item_name]
+            except KeyError:
+                self.item = None
+            return dict(self.item or {})
 
-        metadata = dict(self.item or {})
+        self._worker = ThreadWorker(get_metadata)
+        self._worker.returned.connect(self._got_metadata)
+        self._worker.start()
```

Comment:
Good to do these in threads for anyone who hasn't updated to the latest happi, or in case something else becomes slow or otherwise goes wrong in the future